src/lib/version.cmake

```diff
@@ -0,0 +1,7 @@
+# create or update the version header file with the latest git describe
+execute_process(
+    COMMAND git describe --dirty
+    OUTPUT_VARIABLE GIT_DESCRIBE
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+configure_file(${SRC} ${DST} @ONLY)
```

Comment:
Since we only ever use this cmake script in one place, how about we hardcode SRC and DST (with `${CMAKE_SOURCE_DIR}` prefixes) so that we don't have to pass them in `CMakeLists`?